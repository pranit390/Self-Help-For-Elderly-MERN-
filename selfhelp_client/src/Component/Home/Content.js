import React from 'react';
import styled from 'styled-components'

const Styles = styled.div`
  
   .heading{
       text-align:center;
       margin-top:55px;
       margin-bottom:35px;
       text-decoration:underline;
       font-size:3.5rem;
       font-weight:bold
   }

   .para{
       text-align:center;
       margin-left:15%;
       margin-right:15%;
       color:grey;
       font-size:1.3rem;
   }

   .Container{
    background-color:;
   }
 
`;

function Content(props) {
    return (
        <Styles>
            <>
                <h1 className="heading">The Top Services</h1>
                <div className="Container">
                    <p className="para">India’s leading experts of doctors and nurses under a single roof have come to offer the best home care in Ahmedabad. Healthcare at home, is one of the most sought after requirements in post treatment hospital care. Patients and their relatives wish to be cared for from home, and would prefer being helped and assisted at home. The Self help – Home Care Team, with CIMS Hospital offers quality home care services across Ahmedabad and some of it’s surrounding regions.</p>
                    <br />
                    <p className="para">
                        A growing concern has been providing affordable yet complete care for ailing and recovering patients at home. These patients prefer to be in an environment of their liking, where they have family and friends easily accessible to them. Self Help, makes things easy with our team visiting your home, rather than you visiting us. The importance of Self help services is growing. In the last decade or so, people wish to get services at their door step. Though in healthcare, this is not always possible, Self help – Home Care Unit makes most types of healthcare including preventive checkups possible.
            </p>
                    <br />
                    <p className="para">
                        When the health of a loved one is compromised, all treatment options can seem a bit overwhelming. Often, many patients and their families choose toreceive medical care in the comfort of home. It’s a familiar alternative to the hurried pace of a hospital, nursing home, or assisted living community.
            <br />
            And for that reason Self Help, was created to help make home care easy. Relax at Home. Self help will do the rest !
            </p>
                </div>
            </>
        </Styles>
    );
}

export default Content;