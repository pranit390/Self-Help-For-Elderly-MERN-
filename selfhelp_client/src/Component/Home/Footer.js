import React from 'react';
export const Footer= () => {
 return(
<>
<div className="contact" id="contact">
     <div className="container">
        <div className="content">
          
            <div className="box text wow slideInRight">
                 <h2>Get Connected With Us</h2>
                  <p className="title-p">We Care About You.</p>
                  <div className="info">
                      <ul>
                          <li><span className="fa fa-map-marker"></span> Silicon City, House no 11, New Delhi</li>
                          <li><span className="fa fa-phone"></span> 91 9999999999</li>
                          <li><span className="fa fa-envelope"></span> info@elderlyhelp.com</li>
                      </ul>
                  </div>
                  <div class="social">
                       <a href="https://www.facebook.com/" target="_blank" rel="noreferrer"><span className="fa fa-facebook"></span></a>
                       <a href="https://www.linkedin.com/" target="_blank" rel="noreferrer"><span className="fa fa-linkedin"></span></a>
                       <a href="https://www.skype.com/en/" target="_blank" rel="noreferrer"><span className="fa fa-skype"></span></a>
                       <a href="https://www.youtube.com/" target="_blank" rel="noreferrer"><span className="fa fa-youtube-play"></span></a> 
                  </div>
                  <div className="copyright">
                     
                   <p>Â© Copyright 2021</p>
                         
                   </div>
                  
           </div>
        </div>
     </div>
  </div>
    </>
 )
 }
