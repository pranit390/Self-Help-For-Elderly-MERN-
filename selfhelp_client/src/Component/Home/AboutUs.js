import React, {useEffect} from "react"
import styled from 'styled-components'
import img2 from './images/img2.jpg';
import NavigationBar from '../navbar/navbar.js';
import {Footer} from './Footer'
import '../../css/home/footer.css'

const Styles = styled.div`  
.h1{
   margin-left:40%;
}

.h1 {
    margin-right:20%;
    
    margin-left:20%;
    color: #3d5c5c;
    font-weight: bold;
    
}

.h1 {
    
    color: white;
    margin-left:20%;
    margin-right:20%;
    margin-top: 5%;
    margin-bottom: 2%;
    font-weight: bold;
    
}

.list{
    background-color:black;
    opacity: 0.5;
    margin-left:20%;
    margin-right:20%;
    margin-bottom:5%;
    color: white;
    font-weight: bold;
    font-style: Monospace Lucida Console;
    
}

.para{
    margin-left:10%;
    margin-right:40%;
    font-weight: bold;
    color:white ;
    backgound-color: black;

}

.heading1{
    margin-left:10%;
    font-weight: bold;
    color: white;
}

.section1 {
   
    background:url(${img2}) no-repeat;
    background-size: cover;
  
    background-width:100%
    background-height: 500vh;
    opacity: 0.7;
    margin-top:115px;
    box-shadow: 0px 0px 20px 2px rgba(15, 15, 15, 0.2);
    margin-left: 20px;
    margin-right: 20px;
	border-radius: 10px;

}


`


export const Aboutus=()=>{

    useEffect(()=>{
        window.scrollTo(0, 0);
       },[])


return(
    <>
    <NavigationBar/>
    <Styles>
<div className={"section1"}>
    <h1 className={"heading1"}  >Why Self Help?</h1>
    <p className = {"para"}>
        Self Help offers the best nurses in Ahmedabad and is proud to provide the best doctors
        in Ahmedabad at your doorstep to your home though a few simple steps. Self Help has partnered
        with the 2018 winner for the best hospital in Gujarat – CIMS Hospital to provide the amazing team 
        of doctors and nurses that offer a wide variety of services to your home backed by the excellent 
        standard of care you expect from the best hospital in India, right within the comforts of your own home.
    </p>

    <h1 className={"h1"}>Why Choose Us?</h1>
   <p className={"list"}> Self Help (A UNIT OF CIMS HOSPITAL PVT LTD) – 
       we know what is required to meet your needs at home, even tailoring our services
        to meet individual needs. A level of service which emphasises on a high level of 
        cleanliness and hygiene from our staff to meet and exceed the healthcare industry’s highest standards.
                 
     People often compare services offered by two health care at home providers and find confusion 
    on who to pick. With Self Help powered by CIMS Hospital which has been rated as one of the
     best hospitals for patient experience by patients through online reviews and testimonials, makes
      it easy for you to skip the need to spend time picking and choosing. It’s quite clear that Care 
      At Homes is always going to be your first and last choice. It’s the winner all the time.
        
        
        <ul>
            <li className={"list1"}> Self Help offers 24-hour services, including all holidays and weekends. </li>
            <li className={"list1"}> Self Help offers Highly trained and experienced healthcare professionals ! </li>
            <li className={"list1"}> Self Help offers Regular assessments by management to ensure client satisfaction ! </li>
            <li className={"list1"}> Self Help offers IN HOME consultation ! </li>
        </ul>
    </p>
</div>
</Styles>
<Footer/>
</>
 )
}