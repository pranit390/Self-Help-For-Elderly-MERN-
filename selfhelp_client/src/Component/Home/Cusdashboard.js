import axios from "axios"
import React,{ useEffect, useState } from "react"

import styled from 'styled-components';//For internal styling
//======================================================================================================Importing Images
import cus1 from './customer_dashboard_images/cus1.jpg';
import cus2 from './customer_dashboard_images/cus2.jpg';
import cus3 from './customer_dashboard_images/cus3.jpg';
import img1 from './customer_dashboard_images/img1.jpg';
import service1 from './customer_dashboard_images/service1.jpg';
import service2 from './customer_dashboard_images/service2.jpg';

import walk from './customer_dashboard_images/walk.png'
import morning from './customer_dashboard_images/morning.png';
import evening from './customer_dashboard_images/evening.jpg';
import plumber from './customer_dashboard_images/plumbing-service.png';
import grocery from './customer_dashboard_images/grocery.png';
import housecleaning from './customer_dashboard_images/housecleaning.png';
import service3 from './customer_dashboard_images/medicine-service.png';

import { NavigationBar } from '../navbar/Navigationbar'
import { Footer } from '../Home/Footer.js'

//========================================================================================================Importing APIHELPER 
import APIHelperCustomer from "../../ApiHelper/customerApiHelper"
import { getVolunteerIdBasedOnService } from "../../ApiHelper/volunteerApiHelper.js"
import { createTask } from "../../ApiHelper/customerDashboardApiHelper"
//=========================================================================================================Bootstrap
import {Modal,Button} from 'react-bootstrap';
//===========================================================================================================styling start

//Style introduced using  styled component

import API_URL from "../../ApiHelper/API_URL.js";
const Styles = styled.div`  

.heading2 {
    margin-right:20%;
    background-color:#a8d8f0;
    margin-left:20%;
    color: #3d5c5c;
    padding-top: 20px;
    
}

.heading3 {
    background-color:#a8d8f0;
    color: #3d5c5c;
    margin-left:20%;
    margin-right:20%;
    
}

.list{
    background-color:#e6ffff;
    margin-left:20%;
    margin-right:20%;
    margin-bottom:5%;
    color: grey;
    padding-left:1%;
    padding-right:3%;

    
}

.list1{
    color:grey;
    font-weight: bold;
}

.section1 {
   
    background:url(${img1}) no-repeat;
    background-size: cover;

    background-width:100%
    background-height: 500vh;
    opacity: 0.7;

}

.para{
   
    margin-left:10%;
    margin-right:40%;
    padding-bottom: 100px;
    font-weight: bold;
}

.heading1{
    margin-left:10%;
    padding-top: 100px;
    padding-bottom: 30px;
}

.Choose{
    color:white;
}




 

         
.about{
	padding:80px 0px;
	background-color:#e6f9ff;
}

.about .content{ 
 -webkit-display: flex;
 display: flex;
}
.about .content .box{
	flex:0 0 33.33%;
	max-width: 33.33%;
	padding:15px;
}
.about .content .box .inner{
	background-color: #1aa3ff;
}
.about .content .box:nth-child(2) .inner{
	background-color: 

    #00cccc;
}
.about .content .box .inner .img img{
	width: 100%;
}
.about .content .box .inner .text{
	padding:30px;
}

.about .content .box .inner .text h4{
  font-size: 20px;
  margin:0;
  padding: 0px 0px 15px;
  font-weight: 500;
  color:#ffffff;
 text-transform: capitalize;
  text-align: center;
}
.about .content .box .inner .text p{
	font-size: 15px;
	line-height:20px;
	color:#ffffff;
	text-align: center;
	margin:0;
	font-family: 'Open Sans', sans-serif;
}

.head{
    margin-left:35%;
   padding-bottom: 5%;
   margin-top: 5%;
    color:grey;
}

.heading2{
    margin-top: 3%;
}

.options{
    padding-top:5%;
    padding-right:5%;
    padding-left:5%;
    padding-bottom:5%;
    opacity: 0.5;
}
.options:hover{
    opacity: 1;
    cursor: pointer;
    
}

.service{
    color:white;
}
.service:hover{
    color:black;
    cursor: pointer;
}

.options{
    -moz-box-shadow: 0 0 10px #ccc;
    -webkit-box-shadow: 0 0 10px #ccc;
    box-shadow: 0 0 10px #ccc;
  }


  /*if screen width is less than or equal to 767px then*/
@media(max-width: 767px){
	
  .about .content{
	  flex-wrap: wrap;
  }
  .about .content .box {
	  flex: 0 0 100%;
	  max-width: 100%;
  }
  
  }

 `;


//==================================================================================================================================
//Functional Component 
export const CustomerDashboard = () => {

    //====================================================================PopUp
    const [show, setShow] = useState(false);
  const handleClose = () => setShow(false);
  const handleShow = () =>{ setShow(true);setButtonDisabled(true);}
  //================================================================================POPUp
  const [event,setEvent]=useState(null);
  const[taskName,setTaskName]=useState('');
  const[alertMessage,setAlertMessage]=useState();
  const [buttonDisabled, setButtonDisabled] = useState(true);
  const [privateData,setPrivateData]=useState({});

  useEffect(()=>{
      if(!localStorage.getItem("authToken")){
        window.location.href = "/Login";
      }
      const fetchPrivateDate = async () => {
        const config = {
          headers: {
            "Content-Type": "application/json",
            Authorization: `Bearer ${localStorage.getItem("authToken")}`,
          },
        };
  
        try {
          const  res  = await axios.get(`${API_URL}/auth/getParticularCustomer`, config);
         
          setPrivateData(res.data);
        } catch (error) {
          localStorage.removeItem("authToken");
         // setError("You are not authorized please login");
        }
      };
  
      fetchPrivateDate();
  },[])

    



    //Function called when Services get clicked to confirm
    const message = async (e) => {
        setEvent(e);
        if(e.target.nodeName==="DIV"){
            setAlertMessage(<strong>Your task is {e.target.id}</strong>)  
        }
        if(e.target.nodeName==="IMG"){
        setAlertMessage(<strong>Your task is {e.target.name}</strong>)}
              handleShow();
      }

    const registeredTask=async()=>{

        setButtonDisabled(false);

        const e=event;
        
        //==================================================================================================================Medical Services
        if (e.target.name == "ambulance" || e.target.name == "hospital-visit" || e.target.name == "medicine-delivery") {
            let serviceId = "60478f00c8319b1a9dbd3c05";//medical service ID


            const volunteerId = await getVolunteerIdBasedOnService(serviceId);//Getting volunteer status(available/unavailable).
        
            if (volunteerId !== "not found") {
                  

              
                const details = {
                    Customer_id: privateData._id,
                    Customer_Name: privateData.firstName + " " + privateData.lastName,

                    Customer_address: {
                        address: privateData.address,
                        state: privateData.state,
                        city: privateData.city,
                        pincode: privateData.zipCode
                    },

                    Customer_PhoneNumber: privateData.phoneNumber,



                    Volunteer_Name: volunteerId.firstName + " " + volunteerId.lastName,

                    Volunteer_PhoneNumber: volunteerId.phoneNumber,

                    Volunteer_id:volunteerId,
                    Service_id: serviceId,
                    TaskName: e.target.name

                }
                const task = await createTask(details);
                setAlertMessage(<strong>Your task is registered soon our volunteer will be reaching you for more information
                     </strong>)
            }
            else {
                setAlertMessage(<strong>Sorry! No Volunteers </strong>)
            }
        }

        //===============================================================================================================Assistance in HouseHold
        if (e.target.name == "house-cleaning" || e.target.name == "grocery-delivery" || e.target.name == "plumber-facility") {
            let serviceId = "60479538fe7adc5be8e5ce49";
            const volunteerId = await getVolunteerIdBasedOnService(serviceId)
           
            if (volunteerId !== "not found") {
              

                const details = {
                    Customer_id:privateData._id,
                    Customer_Name: privateData.firstName + " " + privateData.lastName,

                    Customer_address: {
                        address: privateData.address,
                        state: privateData.state,
                        city: privateData.city,
                        pincode: privateData.zipCode
                    },

                    Customer_PhoneNumber: privateData.phoneNumber,



                    Volunteer_Name: volunteerId.firstName + " " + volunteerId.lastName,

                    Volunteer_PhoneNumber: volunteerId.phoneNumber,

                    Volunteer_id: volunteerId,
                    Service_id: serviceId,
                    TaskName: e.target.name

                }
                const task = await createTask(details);
                setAlertMessage(<strong>Your task is registered soon our volunteer will be reaching you for more information
                    </strong>)
            }
            else {
                setAlertMessage(<strong>Sorry! No Volunteers </strong>)
            }
        }


//=========================================================================================================================Transportaion
        if (e.target.name == "morning-walk" || e.target.name == "evening-walk" || e.target.name == "locomotive-service") {
            let serviceId = "60479550fe7adc5be8e5ce4a";
            const volunteerId = await getVolunteerIdBasedOnService(serviceId)
            
            if (volunteerId !== "not found") {
            

                const details = {
                    Customer_id: privateData._id,
                    Customer_Name: privateData.firstName + " " + privateData.lastName,

                    Customer_address: {
                        address: privateData.address,
                        state: privateData.state,
                        city: privateData.city,
                        pincode: privateData.zipCode
                    },

                    Customer_PhoneNumber: privateData.phoneNumber,

                    Volunteer_Name: volunteerId.firstName + " " + volunteerId.lastName,

                    Volunteer_PhoneNumber: volunteerId.phoneNumber,

                    Volunteer_id: volunteerId,
                    Service_id: serviceId,
                    TaskName: e.target.name

                }
                const task = await createTask(details);
                setAlertMessage(<strong>Your task is registered soon our volunteer will be reaching you for more information
                    </strong>)
            }
            else {
                setAlertMessage(<strong>Sorry! No Volunteers </strong>)
            }
        }



        if (e.target.id == "ambulance" || e.target.id == "hospital-visit" || e.target.id == "medicine-delivery") {
            let serviceId = "60478f00c8319b1a9dbd3c05";//medical service ID


            const volunteerId = await getVolunteerIdBasedOnService(serviceId);//Getting volunteer status(available/unavailable).
        
            if (volunteerId !== "not found") {
                  

              
                const details = {
                    Customer_id: privateData._id,
                    Customer_Name: privateData.firstName + " " + privateData.lastName,

                    Customer_address: {
                        address: privateData.address,
                        state: privateData.state,
                        city: privateData.city,
                        pincode: privateData.zipCode
                    },

                    Customer_PhoneNumber: privateData.phoneNumber,



                    Volunteer_Name: volunteerId.firstName + " " + volunteerId.lastName,

                    Volunteer_PhoneNumber: volunteerId.phoneNumber,

                    Volunteer_id:volunteerId,
                    Service_id: serviceId,
                    TaskName: e.target.id

                }
                const task = await createTask(details);
                setAlertMessage(<strong>Your task is registered soon our volunteer will be reaching you for more information
                     </strong>)
            }
            else {
                setAlertMessage(<strong>Sorry! No Volunteers </strong>)
            }
        }

        //===============================================================================================================Assistance in HouseHold
        if (e.target.id == "house-cleaning" || e.target.id== "grocery-delivery" || e.target.id == "plumber-facility") {
            let serviceId = "60479538fe7adc5be8e5ce49";
            const volunteerId = await getVolunteerIdBasedOnService(serviceId)
           
            if (volunteerId !== "not found") {
              

                const details = {
                    Customer_id:privateData._id,
                    Customer_Name: privateData.firstName + " " + privateData.lastName,

                    Customer_address: {
                        address: privateData.address,
                        state: privateData.state,
                        city: privateData.city,
                        pincode: privateData.zipCode
                    },

                    Customer_PhoneNumber: privateData.phoneNumber,



                    Volunteer_Name: volunteerId.firstName + " " + volunteerId.lastName,

                    Volunteer_PhoneNumber: volunteerId.phoneNumber,

                    Volunteer_id: volunteerId,
                    Service_id: serviceId,
                    TaskName: e.target.id

                }
                const task = await createTask(details);
                setAlertMessage(<strong>Your task is registered soon our volunteer will be reaching you for more information
                    </strong>)
            }
            else {
                setAlertMessage(<strong>Sorry! No Volunteers </strong>)
            }
        }


//=========================================================================================================================Transportaion
        if (e.target.id == "morning-walk" || e.target.id== "evening-walk" || e.target.id == "locomotive-service") {
            let serviceId = "60479550fe7adc5be8e5ce4a";
            const volunteerId = await getVolunteerIdBasedOnService(serviceId)
            
            if (volunteerId !== "not found") {
            

                const details = {
                    Customer_id: privateData._id,
                    Customer_Name: privateData.firstName + " " + privateData.lastName,

                    Customer_address: {
                        address: privateData.address,
                        state: privateData.state,
                        city: privateData.city,
                        pincode: privateData.zipCode
                    },

                    Customer_PhoneNumber: privateData.phoneNumber,

                    Volunteer_Name: volunteerId.firstName + " " + volunteerId.lastName,

                    Volunteer_PhoneNumber: volunteerId.phoneNumber,

                    Volunteer_id: volunteerId,
                    Service_id: serviceId,
                    TaskName: e.target.id

                }
                const task = await createTask(details);
                setAlertMessage(<strong>Your task is registered soon our volunteer will be reaching you for more information
                    </strong>)
            }
            else {
                setAlertMessage(<strong>Sorry! No Volunteers </strong>)
            }
        }       
    
    }  
   


    return (
        <>
         <Modal show={show} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>Register Your Task</Modal.Title>
        </Modal.Header>
        <Modal.Body> <p>{alertMessage}</p></Modal.Body>
        <Modal.Footer>
       {
           buttonDisabled && <Button  variant="secondary"   onClick={registeredTask} > Confirm</Button>
       }
          <Button variant="primary" onClick={handleClose}>
            Close
          </Button>
        </Modal.Footer>
      </Modal>


     


            <Styles>
                <NavigationBar />
                <h1 className={"head"}>CUSTOMER DASHBOARD</h1>
                <section className="about" id="about">
                    <div className="container">
                        <div className="content">
                            <div className="box wow bounceInUp">
                                <div className="inner">
                                    <div className="img">
                                        <img src={cus1} alt="img" height="200px " width="100px" />
                                    </div>
                                    <div className="text">
                                        <h4>Medical Services</h4>

                                        <label className={"Choose"} for="medical" >Choose a service:</label>
                                        <div className={"service"} id="ambulance"  onClick={e => message(e)}> <img src={service1} className={"options"} name="ambulance" alt="img" height="80px " width="80px" onClick={(e) => message(e)} /> Ambulance</div>
                                        <div className={"service"}  id="hospital-visit" onClick={e => message(e)}> <img src={service2} className={"options"} name="hospital-visit" alt="img" height="80px " width="80px" onClick={e => message(e)} />Hospital Visit</div>
                                        <div className={"service"} id="medicine-delivery" onClick={e => message(e)}> <img src={service3} className={"options"} name="medicine-delivery" alt="img" height="80px " width="80px" onClick={e => message(e)} /> Medicine delivery</div>

                                    </div>
                                </div>
                            </div>
                            <div className="box wow bounceInUp" >
                                <div className="inner">
                                    <div className="img">
                                        <img src={cus3} alt="img" height="200px " width="100px"/>
                                    </div>
                                    <div className="text">
                                        <h4>Assistance In Household</h4>


                                        <label className={"Choose"}  >Choose a service:</label>
                                        <div className={"service"} id="house-cleaning"  onClick={e => message(e)}> <img src={housecleaning} className={"options"} name="house-cleaning" alt="img" height="80px " width="80px" onClick={e => message(e)} /> House Cleaning</div>
                                        <div className={"service"} id="grocery-delivery"  onClick={e => message(e)}> <img src={grocery} className={"options"} name="grocery-delivery" alt="img" height="80px " width="80px" onClick={e => message(e)} /> Grocery Delivery</div>
                                        <div className={"service"} id="plumber-facility"  onClick={e => message(e)}> <img src={plumber} className={"options"} name="plumber-facility" alt="img" height="80px " width="80px" onClick={e => message(e)} /> Plumber facility</div>







                                    </div>
                                </div>
                            </div>
                            <div class="box wow bounceInUp" data-wow-delay="0.4s">
                                <div class="inner">
                                    <div class="img">
                                        <img src={cus2} alt="img" height="200px " width="100px"  />

                                    </div>

                                    <div class="text">
                                        <h4>Transportation</h4>

                                        <div className="submitdiv">


                                            <label className={"Choose"} for="medical" >Choose a service:</label>
                                            <div className={"service"} id="morning-walk"  onClick={e => message(e)} > <img src={morning} className={"options"} name="morning-walk" alt="img" height="80px " width="80px" onClick={e => message(e)} /> Morning Walk</div>
                                            <div className={"service"} id="evening-walk"  onClick={e => message(e)}> <img src={evening} className={"options"} name="evening-walk" alt="img" height="80px " width="80px" onClick={e => message(e)} /> Evening Walk</div>
                                            <div className={"service"} id="locomotive-service"  onClick={e => message(e)}> <img src={walk} className={"options"} name="locomotive-service" alt="img" height="80px " width="80px" onClick={e => message(e)} /> Locomotive Service</div>


                                            <div class="dropdown">


                                            </div>

                                        </div>



                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </section>




                <h1 className={"heading2"}>Facilities </h1>

                <p className={"list"}>
                    Self Help offers the best care taker and is proud to provide the professionals
                    at your doorstep to your home though a few simple steps. Self Help provide the amazing team
                    of medical and regular task representatives that offer a wide variety of services to your home backed by 
                    the excellent standard of care you expect from the best hospital in India, right within the comforts of your own home.
                    Help with the activities of walking, house cleaning, delivery, or locomotive services, is called
                    personal or custodial care. Home health aides can provide personal care services that range from a few
                    hours a day to around-the-clock live-in care.  They may also provide limited assistance with things
                    such as taking blood pressure or offering medication reminders.
                </p>



            </Styles>
            <Footer />

        </>
    )
}