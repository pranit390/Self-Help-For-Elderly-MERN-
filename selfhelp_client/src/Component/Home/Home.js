import React, { useEffect } from 'react';
import {Carousel,Container} from 'react-bootstrap' 
import corosel2 from '../image/second-corosel.jpg'
import AdminHome_1 from '../image/AdminHome_1.jpg'
import AdminHome_2 from '../image/AdminHome_2.jpg'
import AdminEveningWalk from '../image/AdminEveningWalk.jpg'
import Content from './Content'
import '../../css/home/description.css'
import {Description} from './Description.js'
import NavigationBar from '../navbar/navbar.js';
import {Footer} from './Footer'
import '../../css/home/footer.css'

function Home(props) {

   
    return (
        <>
        <NavigationBar/>
           <Container style={{marginTop: 115}}>
           <Carousel>
                <Carousel.Item interval={3000}>
                    <img
                        className="d-block w-100"
                        src={AdminHome_1}
                        alt="First slide"
                    />
                    <Carousel.Caption>
                        <h1>Self Help</h1>
                        <p>We are here to help you </p>
                    </Carousel.Caption>
                </Carousel.Item>
                <Carousel.Item interval={3000}>
                    <img
                        className="d-block w-100"
                        src={AdminHome_2}
                        alt="First slide"
                    />
                    <Carousel.Caption>
                        <h1>Self Help</h1>
                        <p>We are here to help you </p>
                    </Carousel.Caption>
                </Carousel.Item>
                <Carousel.Item interval={3000}>
                    <img
                        className="d-block w-100"
                        src={AdminEveningWalk}
                        alt="Second slide"
                    />
                    <Carousel.Caption>
                        <h1>Service You love</h1>
                        <p>Always Eager to help whenever you need</p>
                    </Carousel.Caption>
                </Carousel.Item>
                <Carousel.Item interval={3000}>
                    <img
                        className="d-block w-100"
                        src={corosel2}
                        alt="Second slide"
                    />
                    <Carousel.Caption>
                        <h1>Service You love</h1>
                        <p>Always Eager to help whenever you need</p>
                    </Carousel.Caption>
                </Carousel.Item>
                
            </Carousel>
           </Container>
           <Content/>
           <Description/>
           <Footer/>
        </>
    );
}

export default Home;