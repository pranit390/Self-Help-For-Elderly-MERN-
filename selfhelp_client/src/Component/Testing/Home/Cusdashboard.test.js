import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import {shallow} from 'enzyme';
import {CustomerDashboard} from '../../Home/Cusdashboard';




describe("CustomerDashboard", () => {
    it("should render my component", () => {
      const wrapper = shallow(<CustomerDashboard/>);
      expect(wrapper.getElements()).toMatchSnapshot();
    });

   
  });