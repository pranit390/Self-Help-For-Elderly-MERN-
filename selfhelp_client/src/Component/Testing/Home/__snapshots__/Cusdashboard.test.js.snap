// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerDashboard should render my component 1`] = `
Array [
  <React.Fragment>
    <Modal
      animation={true}
      autoFocus={true}
      backdrop={true}
      dialogAs={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      enforceFocus={true}
      keyboard={true}
      onHide={[Function]}
      restoreFocus={true}
      show={false}
    >
      <ModalHeader
        closeButton={true}
        closeLabel="Close"
      >
        <ModalTitle>
          Register Your Task
        </ModalTitle>
      </ModalHeader>
      <ModalBody>
         
        <p />
      </ModalBody>
      <ModalFooter>
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          variant="secondary"
        >
           Confirm
        </Button>
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          variant="primary"
        >
          Close
        </Button>
      </ModalFooter>
    </Modal>
    <styled.div>
      <NavigationBar />
      <h1
        className="head"
      >
        CUSTOMER DASHBOARD
      </h1>
      <section
        className="about"
        id="about"
      >
        <div
          className="container"
        >
          <div
            className="content"
          >
            <div
              className="box wow bounceInUp"
            >
              <div
                className="inner"
              >
                <div
                  className="img"
                >
                  <img
                    alt="img"
                    height="200px "
                    src="cus1.jpg"
                    width="100px"
                  />
                </div>
                <div
                  className="text"
                >
                  <h4>
                    Medical Services
                  </h4>
                  <label
                    className="Choose"
                    for="medical"
                  >
                    Choose a service:
                  </label>
                  <div
                    className="service"
                    id="ambulance"
                    onClick={[Function]}
                  >
                     
                    <img
                      alt="img"
                      className="options"
                      height="80px "
                      name="ambulance"
                      onClick={[Function]}
                      src="service1.jpg"
                      width="80px"
                    />
                     Ambulance
                  </div>
                  <div
                    className="service"
                    id="hospital-visit"
                    onClick={[Function]}
                  >
                     
                    <img
                      alt="img"
                      className="options"
                      height="80px "
                      name="hospital-visit"
                      onClick={[Function]}
                      src="service2.jpg"
                      width="80px"
                    />
                    Hospital Visit
                  </div>
                  <div
                    className="service"
                    id="medicine-delivery"
                    onClick={[Function]}
                  >
                     
                    <img
                      alt="img"
                      className="options"
                      height="80px "
                      name="medicine-delivery"
                      onClick={[Function]}
                      src="medicine-service.png"
                      width="80px"
                    />
                     Medicine delivery
                  </div>
                </div>
              </div>
            </div>
            <div
              className="box wow bounceInUp"
            >
              <div
                className="inner"
              >
                <div
                  className="img"
                >
                  <img
                    alt="img"
                    height="200px "
                    src="cus3.jpg"
                    width="100px"
                  />
                </div>
                <div
                  className="text"
                >
                  <h4>
                    Assistance In Household
                  </h4>
                  <label
                    className="Choose"
                  >
                    Choose a service:
                  </label>
                  <div
                    className="service"
                    id="house-cleaning"
                    onClick={[Function]}
                  >
                     
                    <img
                      alt="img"
                      className="options"
                      height="80px "
                      name="house-cleaning"
                      onClick={[Function]}
                      src="housecleaning.png"
                      width="80px"
                    />
                     House Cleaning
                  </div>
                  <div
                    className="service"
                    id="grocery-delivery"
                    onClick={[Function]}
                  >
                     
                    <img
                      alt="img"
                      className="options"
                      height="80px "
                      name="grocery-delivery"
                      onClick={[Function]}
                      src="grocery.png"
                      width="80px"
                    />
                     Grocery Delivery
                  </div>
                  <div
                    className="service"
                    id="plumber-facility"
                    onClick={[Function]}
                  >
                     
                    <img
                      alt="img"
                      className="options"
                      height="80px "
                      name="plumber-facility"
                      onClick={[Function]}
                      src="plumbing-service.png"
                      width="80px"
                    />
                     Plumber facility
                  </div>
                </div>
              </div>
            </div>
            <div
              class="box wow bounceInUp"
              data-wow-delay="0.4s"
            >
              <div
                class="inner"
              >
                <div
                  class="img"
                >
                  <img
                    alt="img"
                    height="200px "
                    src="cus2.jpg"
                    width="100px"
                  />
                </div>
                <div
                  class="text"
                >
                  <h4>
                    Transportation
                  </h4>
                  <div
                    className="submitdiv"
                  >
                    <label
                      className="Choose"
                      for="medical"
                    >
                      Choose a service:
                    </label>
                    <div
                      className="service"
                      id="morning-walk"
                      onClick={[Function]}
                    >
                       
                      <img
                        alt="img"
                        className="options"
                        height="80px "
                        name="morning-walk"
                        onClick={[Function]}
                        src="morning.png"
                        width="80px"
                      />
                       Morning Walk
                    </div>
                    <div
                      className="service"
                      id="evening-walk"
                      onClick={[Function]}
                    >
                       
                      <img
                        alt="img"
                        className="options"
                        height="80px "
                        name="evening-walk"
                        onClick={[Function]}
                        src="evening.jpg"
                        width="80px"
                      />
                       Evening Walk
                    </div>
                    <div
                      className="service"
                      id="locomotive-service"
                      onClick={[Function]}
                    >
                       
                      <img
                        alt="img"
                        className="options"
                        height="80px "
                        name="locomotive-service"
                        onClick={[Function]}
                        src="walk.png"
                        width="80px"
                      />
                       Locomotive Service
                    </div>
                    <div
                      class="dropdown"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <h1
        className="heading2"
      >
        Facilities 
      </h1>
      <p
        className="list"
      >
        Self Help offers the best care taker and is proud to provide the professionals at your doorstep to your home though a few simple steps. Self Help provide the amazing team of medical and regular task representatives that offer a wide variety of services to your home backed by the excellent standard of care you expect from the best hospital in India, right within the comforts of your own home. Help with the activities of walking, house cleaning, delivery, or locomotive services, is called personal or custodial care. Home health aides can provide personal care services that range from a few hours a day to around-the-clock live-in care.  They may also provide limited assistance with things such as taking blood pressure or offering medication reminders.
      </p>
    </styled.div>
    <Footer />
  </React.Fragment>,
]
`;
